src/test/java/se/sundsvall/byggrintegrator/service/LegalIdUtilityTest.java

```diff
@@ -0,0 +1,41 @@
+package se.sundsvall.byggrintegrator.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static se.sundsvall.byggrintegrator.service.LegalIdUtility.addHyphen;
+
+import org.junit.jupiter.api.Test;
+
+class LegalIdUtilityTest {
+
+	@Test
+	void testAddHyphenOnNull() {
+		assertThat(LegalIdUtility.addHyphen(null)).isNull();
+	}
+
+	@Test
+	void testAddHyphenOnEmptyString() {
+		assertThat(addHyphen("")).isEmpty();
+	}
+
+	@Test
+	void testAddHyphenOnStringWithHyphen() {
+		assertThat(addHyphen("12345-67890")).isEqualTo("12345-67890");
+	}
+
+	@Test
```

Comment:
`@ParameterizedTest` with some kind of key-value parameter list (i.e. "input" -> "expected") ?